tests/specs/lemmas.k

```diff
@@ -267,6 +267,12 @@ module LEMMAS-JAVA [symbolic, kast]
     rule #buf( 32, DATA ) ==K BA:ByteArray     => DATA ==Int #asInteger(BA)
       requires #sizeByteArray(BA) <=Int 32                                      andBool  #isConcrete(BA) [simplification]
 
+    //For #bufStrict simplification in benchmarks
+    rule 0 <Int 2 ^Int I           => true
+      requires 0 <=Int I                                                        [simplification]
+    rule 0 <=Int I *Int 8          => 0 <=Int I                                 [simplification]
```

Comment:
Is it required that we rewrite the actual condition? Could we get away with instead `rule 0 <=Int I *Int 8 => true requires 0 <=Int I [simplification]`, so that we don't lose the original constraint form? Maybe some other lemmas are based on the `I *Int 8` showing up on the RHS of the `<=Int`.

Reply:
Rules with no side conditions are preferred for performance. Once they match the are only evaluated once. A rule with side condition that doesn't match will be tried by the engine every time term constraint changes. I can rewrite as you suggested, please confirm.